website/docs: enhance customization docs (#15081)
goauthentik/authentik: website/docs/customize/interfaces/user-admin/customization.mdx

---
title: Customize the User and Admin interfaces
---

> The authentik User interface and Admin interface can both be customized through attributes, defined in [Brands](../../../sys-mgmt/brands.md) under **Other global settings > Attributes**.

Reviewer: ```suggestion
Both the authentik User interface and Admin interface can be customized using attributes configured in the [Brands](../../../sys-mgmt/brands.md) section, found under **Other global settings > Attributes**.
```

Most settings that are defined in the Brand's attributes are manifested in _both_ the User interface and the Admin interface, except for those marked below as either only for the Admin interface or only for the User interface.

## Custom settings

The following settings are grouped by:

- `enabledFeatures` settings
- general configurations

### Enabling/disabling features

> The following features can be enabled/disabled using attributes defined on the Brand. By default, all of them are enabled; to disable a feature, set the value to `false`, as shown in the screenshot below.

Reviewer: ```suggestion
The features listed below can be enabled or disabled through attributes set on the Brand. By default, all features are enabled. To disable a specific feature, set its value to `false`, as shown in the screenshot below.
```

#### `settings.enabledFeatures.apiDrawer`

    Display the API Request drawer in the upper tool bar.

#### `settings.enabledFeatures.notificationDrawer`

> Display the Notification drawer in the upper tool bar.

Reviewer 2: Indentation here doesn't match the other lines

Author: fixed, eagle eyes!

> Display the Notification drawer in the upper tool bar.

Reviewer 2: Seems like prettier really doesn't like this line being indented. Might need to remove the indentation from the other lines if prettier is particularly stubborn.

Author: I'll try un-indenting them and see if it helps build pass. Then we can work on better formatting later.

#### `settings.enabledFeatures.settings`

    Display the Settings link in the upper tool bar.

#### `settings.enabledFeatures.applicationEdit` (User interface only)

    Application edit in library (only shown when user is superuser)

#### `settings.enabledFeatures.search`

    Display the Search bar in the upper tool bar.

### General configurations

#### `settings.navbar.userDisplay`

> Configure what is shown in the top right corner. Defaults to `username`. Choices: `username`, `name`, `email`

Reviewer 2: ```suggestion
Configure what is shown in the top right corner. Defaults to `username`. Available options: `username`, `name`, `email`
```

#### `settings.theme.base`

> Configure the base color scheme, or dark/light mode. Defaults to `automatic`, which switches between dark and light mode based on the users' browsers' preference. Choices: `automatic`, `dark`, `light`.

Reviewer: also "Available options:" vs "Available options are:"

**Example**:

```
settings:
  theme:
    base: dark
```

#### `settings.theme.background`

Optional CSS that is applied to the background of the User interface, for example to set a custom background color, gradient, or image.

```yaml
settings:
    theme:
        background: >
            background: url('https://picsum.photos/1920/1080');
            filter: blur(8px);
            background-position: center;
            background-repeat: no-repeat;
            background-size: cover;
```

#### `settings.layout.type` (User interface only)

> Which layout to use for the _My applications_ view. Defaults to `row`. Choices: `row`, `2-column`, `3-column`

Reviewer 2: Maybe a discussion to have elsewhere. "The My Applications view" seems a bit clunky when referring to this. Maybe we could find a better term and standardize it across documentation and the UI. "Application dashboard" for example.

Author: Ideally one wants to use the exact name/label on the actual UI.. and that page says "My applications." This line is from many years ago (you can tell because it is italized), so ew need to update that for sure.. I'd update to say "My applications page" instead if "view".

But yeah if it ever changed in the UI to be Application dashboard we would want to chnage the docs too.

> Which layout to use for the _My applications_ view. Defaults to `row`. Choices: `row`, `2-column`, `3-column`

Reviewer 2: ```suggestion
Which layout to use for the _My applications_ view. Defaults to `row`. Available options: `row`, `2-column`, `3-column`
```
